pkg/utils/build/task_results.go

```diff
@@ -36,15 +36,44 @@ type Vulnerabilities struct {
 	Low      int `json:"low"`
 }
 
+type PipelineBuildInfo struct {
+	runtime  string
+	strategy string
+}
+
+func GetPipelineBuildInfo(pr *pipeline.PipelineRun) PipelineBuildInfo {
+	var runtime, strategy string
+	var exists bool
+	labels := pr.GetLabels()
+	runtime, exists = labels["pipelines.openshift.io/runtime"]
+	Expect(exists).Should(BeTrue())
+	strategy, exists = labels["pipelines.openshift.io/strategy"]
+	Expect(exists).Should(BeTrue())
+	return PipelineBuildInfo{
+		runtime:  runtime,
+		strategy: strategy,
+	}
+}
+
+func IsDockerBuild(pr *pipeline.PipelineRun) bool {
+	info := GetPipelineBuildInfo(pr)
+	return info.runtime == "generic" && info.strategy == "docker"
+}
+
+func IsFBCBuild(pr *pipeline.PipelineRun) bool {
+	info := GetPipelineBuildInfo(pr)
+	return info.runtime == "fbc" && info.strategy == "fbc"
+}
+
 func ValidateBuildPipelineTestResults(pipelineRun *pipeline.PipelineRun, c crclient.Client) error {
 	for _, taskName := range taskNames {
 		// The inspect-image task is only required for FBC pipelines which we can infer by the component name
-		prLabels := pipelineRun.GetLabels()
-		componentName := prLabels["appstudio.openshift.io/component"]
-		if taskName == "inspect-image" && !strings.HasPrefix(strings.ToLower(componentName), "fbc-") {
+		isFBCBuild := IsFBCBuild(pipelineRun)
+
+		if !isFBCBuild && taskName == "inspect-image" {
 			continue
 		}
-		if taskName == "clair-scan" && !strings.HasPrefix(strings.ToLower(componentName), "fbc-") {
+		if isFBCBuild && taskName == "clair-scan" {
```

Comment:
I think we need to also ignore `clamav-scan`.

Reply:
Will add, thank you